compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/java/KSClassDeclarationJavaImpl.kt

```diff
@@ -92,10 +93,11 @@ class KSClassDeclarationJavaImpl private constructor(val psi: PsiClass) : KSClas
     }
 
     override val declarations: List<KSDeclaration> by lazy {
-        psi.fields.map { KSPropertyDeclarationJavaImpl.getCached(it) } +
+        (psi.fields.map { KSPropertyDeclarationJavaImpl.getCached(it) } +
                 psi.innerClasses.map { KSClassDeclarationJavaImpl.getCached(it) } +
                 psi.constructors.map { KSFunctionDeclarationJavaImpl.getCached(it) } +
-                psi.methods.map { KSFunctionDeclarationJavaImpl.getCached(it) }
+                psi.methods.map { KSFunctionDeclarationJavaImpl.getCached(it) })
+                .distinct()
```

Comment:
Do we know where the duplication comes from?

Reply:
it comes from constructors and methods both returning same psi.